### PR TITLE
Impl. Subpages links in #7 

### PR DESCRIPTION
**Pull request by WooDzu**
This should do for now. Yet need to test on a real-word project.
Tested on virgin Joomla 4.1

### CODE REVIEW

**Comment by WooDzu** [comment]
I've been able to do some extensive tests today with a few production websites.
@juliopontes it would be great if you could take a look at the PR and and advise.

Below are results of what I found works and what doesn't at the moment.
`OLD $live_site` is what was set in $live_site configuration.php on original site
`NEW $base_url` is what was set in component's configuration, which will be updated into `<base>` tag for generated content.

```
OLD $live_site => NEW $base_url                    | Works under NEW $base_url
-----------------------
 '' => '/'                                         | OK
 '' => '/new/'                                     | OK
 '' => 'http://new.com/'                           | OK
 '' => 'http://new.com/static/'                    | OK

 'http://old.com/' => '/'                          | OK
 'http://old.com/' => '/static/'                   | OK
 'http://old.com/' => 'http://new.com/'            | OK
 'http://old.com/' => 'http://new.com/static/'     | OK

 'http://old.com/sub' => '/'                       | OK
 'http://old.com/sub' => '/static/'                | NO
 'http://old.com/sub' => 'http://new.com/'         | OK
 'http://old.com/sub' => 'http://new.com/static/'  | NO
```
TLDR; is that currently a joomla website running from a subfolder (eg `/sub`) doesn't not work properly for generating content for another domain with subfolder. I do plan to look into this as it may come in handy in some cases.

I've also found that links to some static local files (pdf's, videos, zip) aren't handled properly.
It would attempt to generate a 'page' for them instead of copying directly.
This may deserve a separate ticket though as I believe it'd been in the code before.

**Comment by juliopontes** [comment]
Thanks
